### PR TITLE
Remove eyre and use thiserror

### LINKED ISSUE #2: Use our own error type instead of eyre

### LINKED ISSUE #12: Remove formatting options
Formatting options causes the Rust stable toolchain to raise warning when formatting, but it does not justify the use of the nightly toolchain.

https://github.com/Cosmian/crypto_base/blob/6880838bd10e9386b76e2e9ba454d2213c894d24/rustfmt.toml#L1

### PR DESCRIPTION
**Pull request by ThibsG**
fixes #2 
fixes #12

### CODE REVIEW

File: src/symmetric_crypto/ff1.rs [hunk 1]
```diff
@@ -88,26 +92,27 @@ impl RebasedInput {
...
-        );
+        if alphabet.len() ^ stripped_input.len() >= RECOMMENDED_THRESHOLD {
+            return Err(CryptoBaseError::InvalidSize( format!("Given alphabet length ({}), plaintext is too short. Plaintext length should be at least {}", alphabet.len(),(RECOMMENDED_THRESHOLD as f32).log(alphabet.len() as f32))));
+        }
 
         // Fill the mapping between original representation ("ABCDEFG...") and
         // new base representation ("01234567...")
         let alphabet = alphabet.chars().sorted().unique().collect::<Vec<_>>();
-        for i in 0_u8..u8::try_from(alphabet.len())? {
+        let len = u8::try_from(alphabet.len()).map_err(|e| {
```
Comment (Manuthor) [inline src/symmetric_crypto/ff1.rs:109]: not a bad idea to enrich error messages

**Review by Manuthor — COMMENTED** [review]
well not much to say, well done!
Running `cargo +nightly clippy --all-features --workspace --fix -Z unstable-options --allow-dirty --allow-staged -- --no-deps -D warnings -W clippy::nursery` give interesting simplifications, could you give it a try?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: -- -D warnings
+          args: --all-targets -- -D warnings
```
[the remark this answers is not in the document]
Comment (ThibsG) [reply .github/workflows/ci.yml:29]: it cannot compile `libsodium` feature in CI, didn't dig in, maybe we could add `libsodium-dev` to try

File: src/asymmetric/ristretto/mod.rs [hunk 3]
```diff
@@ -576,7 +578,7 @@ mod test {
             &key_pair.private_key,
             &enc_sym_key,
         );
-        println!("decrypted_key: {:?}", decrypted_key);
+        // println!("decrypted_key: {:?}", decrypted_key);
```
[the remark this answers is not in the document]
Comment (ThibsG) [reply src/asymmetric/ristretto/mod.rs:581]: old code?

[the remark this answers is not in the document]
Comment (ThibsG) [reply src/asymmetric/ristretto/mod.rs:581]: not a leftover, it was not commented at the origin so I commented it out

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ThibsG) [reply .github/workflows/ci.yml:29]: fixed in https://github.com/Cosmian/crypto_base/pull/22

**Pull request merged**